server/Server.js

```diff
@@ -58,6 +59,35 @@ app.post("/gauth", (req, res) => {
     .catch(err => errorHandler(err, res));
 });
 
+app.post("/auth", (req, res) => {
+  return UserAuth.validatePassword(req.body.email, req.body.password)
+  .then((userData) => {
+    if (!userData) {
+      return res.sendStatus(400);
+    } else {
+      return res.send(userData);
+    }
+  })
+  .catch(err => errorHandler(err, res));
+});
+
+app.post("/addUser", (req, res) => {
+  UserAuth.addUser(req.body.name, req.body.email, req.body.password)
+  .then((result) => {
+    console.log(result);
+    if (!result) {
+      return res.sendStatus(400);
+    } else if (result===1){
+      return res.send({error: "Email already exists! Please login."});
```

Comment:
Shouldn't send, because by sending, this returns a 200 error. Maybe return a custom error code like this: 

```
res.status(4XX).send(...)
```
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status for the right status code to use

Reply:
Custom error status(4xx).send(err) Does not seem to work, and it just returns undefined. This is the only workaround that did actually send the status.

Reply:
I'll approve it for now, but I think it's because you're supposed to handle non-2XX errors in the catch clause